src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/FlowEntityPath.groovy

```diff
@@ -55,5 +56,17 @@ class FlowEntityPath {
             isProtected ? flowPath.protectedPath.reverse.nodes.toPathNode() : flowPath.path.reverse.nodes.toPathNode()
         }
     }
+
+    List<Isl> getMainPathInvolvedIsls(Direction direction = Direction.FORWARD) {
+        direction == Direction.FORWARD ? flowPath.path.forward.getInvolvedIsls() : flowPath.path.reverse.getInvolvedIsls()
+    }
+
+    List<Isl> getProtectedPathInvolvedIsls(Direction direction = Direction.FORWARD) {
+        direction == Direction.FORWARD ? flowPath.protectedPath.forward.getInvolvedIsls() :  flowPath.protectedPath.reverse.getInvolvedIsls()
+    }
```

Comment:
You could do something like this not to repeat the logic:

    List<Isl> getMainPathInvolvedIsls(Direction direction = Direction.FORWARD) {
                getPathInvolvedIsls(direction, flowPath.path)
    }

    List<Isl> getProtectedPathInvolvedIsls(Direction direction = Direction.FORWARD) {
        getPathInvolvedIsls(direction, flowPath.protectedPath)
    }
    
     List<Isl> getPathInvolvedIsls(Direction direction = Direction.FORWARD, Path path) {
          direction == Direction.FORWARD ? path.forward.getInvolvedIsls() : path.reverse.getInvolvedIsls()
    }